// @openid3/contracts/types/factories/AccountAdminBase__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Interface, type ContractRunner } from "ethers";
import type {
  AccountAdminBase,
  AccountAdminBaseInterface,
} from "../AccountAdminBase";

const _abi = [
  {
    inputs: [],
    name: "OnlyAdminAllowed",
    type: "error",
  },
  {
    inputs: [
      {
        internalType: "bytes4",
        name: "interfaceId",
        type: "bytes4",
      },
    ],
    name: "supportsInterface",
    outputs: [
      {
        internalType: "bool",
        name: "",
        type: "bool",
      },
    ],
    stateMutability: "pure",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "bytes32",
        name: "challenge",
        type: "bytes32",
      },
      {
        internalType: "bytes",
        name: "signature",
        type: "bytes",
      },
    ],
    name: "validateSignature",
    outputs: [
      {
        internalType: "uint256",
        name: "validationData",
        type: "uint256",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
] as const;

export class AccountAdminBase__factory {
  static readonly abi = _abi;
  static createInterface(): AccountAdminBaseInterface {
    return new Interface(_abi) as AccountAdminBaseInterface;
  }
  static connect(
    address: string,
    runner?: ContractRunner | null
  ): AccountAdminBase {
    return new Contract(address, _abi, runner) as unknown as AccountAdminBase;
  }
}
