// @openid3/contracts/types/factories/IAccountValidator__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Interface, type ContractRunner } from "ethers";
import type {
  IAccountValidator,
  IAccountValidatorInterface,
} from "../IAccountValidator";

const _abi = [
  {
    inputs: [
      {
        internalType: "bytes32",
        name: "challenge",
        type: "bytes32",
      },
      {
        internalType: "bytes",
        name: "signature",
        type: "bytes",
      },
    ],
    name: "validateSignature",
    outputs: [
      {
        internalType: "uint256",
        name: "validationData",
        type: "uint256",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
] as const;

export class IAccountValidator__factory {
  static readonly abi = _abi;
  static createInterface(): IAccountValidatorInterface {
    return new Interface(_abi) as IAccountValidatorInterface;
  }
  static connect(
    address: string,
    runner?: ContractRunner | null
  ): IAccountValidator {
    return new Contract(address, _abi, runner) as unknown as IAccountValidator;
  }
}
