// @openid3/contracts/types/factories/IOpenId3Account__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Interface, type ContractRunner } from "ethers";
import type {
  IOpenId3Account,
  IOpenId3AccountInterface,
} from "../IOpenId3Account";

const _abi = [
  {
    inputs: [],
    name: "getAdmin",
    outputs: [
      {
        internalType: "address",
        name: "",
        type: "address",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
  {
    inputs: [],
    name: "getMode",
    outputs: [
      {
        internalType: "uint8",
        name: "",
        type: "uint8",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
  {
    inputs: [],
    name: "getOperator",
    outputs: [
      {
        internalType: "address",
        name: "",
        type: "address",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "bytes",
        name: "adminData",
        type: "bytes",
      },
      {
        internalType: "bytes",
        name: "operatorData",
        type: "bytes",
      },
      {
        internalType: "bytes",
        name: "metadata",
        type: "bytes",
      },
    ],
    name: "initialize",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
] as const;

export class IOpenId3Account__factory {
  static readonly abi = _abi;
  static createInterface(): IOpenId3AccountInterface {
    return new Interface(_abi) as IOpenId3AccountInterface;
  }
  static connect(
    address: string,
    runner?: ContractRunner | null
  ): IOpenId3Account {
    return new Contract(address, _abi, runner) as unknown as IOpenId3Account;
  }
}
