// @openid3/contracts/types/factories/IRsaCertRegistration__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Interface, type ContractRunner } from "ethers";
import type {
  IRsaCertRegistration,
  IRsaCertRegistrationInterface,
} from "../IRsaCertRegistration";

const _abi = [
  {
    inputs: [
      {
        internalType: "bytes32",
        name: "kidHash",
        type: "bytes32",
      },
    ],
    name: "getCert",
    outputs: [
      {
        internalType: "address",
        name: "",
        type: "address",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
] as const;

export class IRsaCertRegistration__factory {
  static readonly abi = _abi;
  static createInterface(): IRsaCertRegistrationInterface {
    return new Interface(_abi) as IRsaCertRegistrationInterface;
  }
  static connect(
    address: string,
    runner?: ContractRunner | null
  ): IRsaCertRegistration {
    return new Contract(
      address,
      _abi,
      runner
    ) as unknown as IRsaCertRegistration;
  }
}
