// @openid3/contracts/types/factories/IRsaCert__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Interface, type ContractRunner } from "ethers";
import type { IRsaCert, IRsaCertInterface } from "../IRsaCert";

const _abi = [
  {
    inputs: [],
    name: "E",
    outputs: [
      {
        internalType: "bytes",
        name: "",
        type: "bytes",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
  {
    inputs: [],
    name: "N",
    outputs: [
      {
        internalType: "bytes",
        name: "",
        type: "bytes",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
  {
    inputs: [],
    name: "getIssuer",
    outputs: [
      {
        internalType: "bytes32",
        name: "",
        type: "bytes32",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
] as const;

export class IRsaCert__factory {
  static readonly abi = _abi;
  static createInterface(): IRsaCertInterface {
    return new Interface(_abi) as IRsaCertInterface;
  }
  static connect(address: string, runner?: ContractRunner | null): IRsaCert {
    return new Contract(address, _abi, runner) as unknown as IRsaCert;
  }
}
