// @openid3/contracts/types/IOpenIdZkVerifier.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
import type {
  BaseContract,
  BigNumberish,
  BytesLike,
  FunctionFragment,
  Result,
  Interface,
  AddressLike,
  ContractRunner,
  ContractMethod,
  Listener,
} from "ethers";
import type {
  TypedContractEvent,
  TypedDeferredTopicFilter,
  TypedEventLog,
  TypedListener,
  TypedContractMethod,
} from "./common";

export type UserOperationStruct = {
  sender: AddressLike;
  nonce: BigNumberish;
  initCode: BytesLike;
  callData: BytesLike;
  callGasLimit: BigNumberish;
  verificationGasLimit: BigNumberish;
  preVerificationGas: BigNumberish;
  maxFeePerGas: BigNumberish;
  maxPriorityFeePerGas: BigNumberish;
  paymasterAndData: BytesLike;
  signature: BytesLike;
};

export type UserOperationStructOutput = [
  sender: string,
  nonce: bigint,
  initCode: string,
  callData: string,
  callGasLimit: bigint,
  verificationGasLimit: bigint,
  preVerificationGas: bigint,
  maxFeePerGas: bigint,
  maxPriorityFeePerGas: bigint,
  paymasterAndData: string,
  signature: string
] & {
  sender: string;
  nonce: bigint;
  initCode: string;
  callData: string;
  callGasLimit: bigint;
  verificationGasLimit: bigint;
  preVerificationGas: bigint;
  maxFeePerGas: bigint;
  maxPriorityFeePerGas: bigint;
  paymasterAndData: string;
  signature: string;
};

export type OpenIdZkProofPublicInputStruct = {
  kidHash: BytesLike;
  iat: string;
  jwtHeaderAndPayloadHash: BytesLike;
  jwtSignature: BytesLike;
};

export type OpenIdZkProofPublicInputStructOutput = [
  kidHash: string,
  iat: string,
  jwtHeaderAndPayloadHash: string,
  jwtSignature: string
] & {
  kidHash: string;
  iat: string;
  jwtHeaderAndPayloadHash: string;
  jwtSignature: string;
};

export interface IOpenIdZkVerifierInterface extends Interface {
  getFunction(
    nameOrSignature:
      | "getIdHash"
      | "getVerificationKeyOfIdp"
      | "verify"
      | "verifyAggregated"
  ): FunctionFragment;

  encodeFunctionData(
    functionFragment: "getIdHash",
    values: [AddressLike]
  ): string;
  encodeFunctionData(
    functionFragment: "getVerificationKeyOfIdp",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "verify",
    values: [UserOperationStruct, OpenIdZkProofPublicInputStruct, BytesLike]
  ): string;
  encodeFunctionData(
    functionFragment: "verifyAggregated",
    values: [UserOperationStruct[], OpenIdZkProofPublicInputStruct[], BytesLike]
  ): string;

  decodeFunctionResult(functionFragment: "getIdHash", data: BytesLike): Result;
  decodeFunctionResult(
    functionFragment: "getVerificationKeyOfIdp",
    data: BytesLike
  ): Result;
  decodeFunctionResult(functionFragment: "verify", data: BytesLike): Result;
  decodeFunctionResult(
    functionFragment: "verifyAggregated",
    data: BytesLike
  ): Result;
}

export interface IOpenIdZkVerifier extends BaseContract {
  connect(runner?: ContractRunner | null): IOpenIdZkVerifier;
  waitForDeployment(): Promise<this>;

  interface: IOpenIdZkVerifierInterface;

  queryFilter<TCEvent extends TypedContractEvent>(
    event: TCEvent,
    fromBlockOrBlockhash?: string | number | undefined,
    toBlock?: string | number | undefined
  ): Promise<Array<TypedEventLog<TCEvent>>>;
  queryFilter<TCEvent extends TypedContractEvent>(
    filter: TypedDeferredTopicFilter<TCEvent>,
    fromBlockOrBlockhash?: string | number | undefined,
    toBlock?: string | number | undefined
  ): Promise<Array<TypedEventLog<TCEvent>>>;

  on<TCEvent extends TypedContractEvent>(
    event: TCEvent,
    listener: TypedListener<TCEvent>
  ): Promise<this>;
  on<TCEvent extends TypedContractEvent>(
    filter: TypedDeferredTopicFilter<TCEvent>,
    listener: TypedListener<TCEvent>
  ): Promise<this>;

  once<TCEvent extends TypedContractEvent>(
    event: TCEvent,
    listener: TypedListener<TCEvent>
  ): Promise<this>;
  once<TCEvent extends TypedContractEvent>(
    filter: TypedDeferredTopicFilter<TCEvent>,
    listener: TypedListener<TCEvent>
  ): Promise<this>;

  listeners<TCEvent extends TypedContractEvent>(
    event: TCEvent
  ): Promise<Array<TypedListener<TCEvent>>>;
  listeners(eventName?: string): Promise<Array<Listener>>;
  removeAllListeners<TCEvent extends TypedContractEvent>(
    event?: TCEvent
  ): Promise<this>;

  getIdHash: TypedContractMethod<[account: AddressLike], [string], "view">;

  getVerificationKeyOfIdp: TypedContractMethod<[], [string], "view">;

  verify: TypedContractMethod<
    [
      op: UserOperationStruct,
      input: OpenIdZkProofPublicInputStruct,
      proof: BytesLike
    ],
    [void],
    "nonpayable"
  >;

  verifyAggregated: TypedContractMethod<
    [
      ops: UserOperationStruct[],
      inputs: OpenIdZkProofPublicInputStruct[],
      aggregatedProof: BytesLike
    ],
    [void],
    "nonpayable"
  >;

  getFunction<T extends ContractMethod = ContractMethod>(
    key: string | FunctionFragment
  ): T;

  getFunction(
    nameOrSignature: "getIdHash"
  ): TypedContractMethod<[account: AddressLike], [string], "view">;
  getFunction(
    nameOrSignature: "getVerificationKeyOfIdp"
  ): TypedContractMethod<[], [string], "view">;
  getFunction(
    nameOrSignature: "verify"
  ): TypedContractMethod<
    [
      op: UserOperationStruct,
      input: OpenIdZkProofPublicInputStruct,
      proof: BytesLike
    ],
    [void],
    "nonpayable"
  >;
  getFunction(
    nameOrSignature: "verifyAggregated"
  ): TypedContractMethod<
    [
      ops: UserOperationStruct[],
      inputs: OpenIdZkProofPublicInputStruct[],
      aggregatedProof: BytesLike
    ],
    [void],
    "nonpayable"
  >;

  filters: {};
}
