// @openid3/contracts/dist/types/factories/IPlonkVerifier__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IPlonkVerifier__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "bytes",
                name: "proof",
                type: "bytes",
            },
            {
                internalType: "uint256[]",
                name: "publicInput",
                type: "uint256[]",
            },
        ],
        name: "verify",
        outputs: [
            {
                internalType: "bool",
                name: "",
                type: "bool",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
class IPlonkVerifier__factory {
    static createInterface() {
        return new ethers_1.Interface(_abi);
    }
    static connect(address, runner) {
        return new ethers_1.Contract(address, _abi, runner);
    }
}
exports.IPlonkVerifier__factory = IPlonkVerifier__factory;
IPlonkVerifier__factory.abi = _abi;
