// @openid3/contracts/dist/types/factories/IRsaCertRegistration__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IRsaCertRegistration__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "kidHash",
                type: "bytes32",
            },
        ],
        name: "getCert",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
class IRsaCertRegistration__factory {
    static createInterface() {
        return new ethers_1.Interface(_abi);
    }
    static connect(address, runner) {
        return new ethers_1.Contract(address, _abi, runner);
    }
}
exports.IRsaCertRegistration__factory = IRsaCertRegistration__factory;
IRsaCertRegistration__factory.abi = _abi;
