// @openid3/contracts/dist/types/factories/MerkleProof__factory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MerkleProof__factory = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [],
        name: "MerkleProofInvalidMultiproof",
        type: "error",
    },
];
const _bytecode = "0x60808060405234601757603a9081601d823930815050f35b600080fdfe600080fdfea2646970667358221220937d9eb131d2f9a0e625bc2d04ec20d8fa470b4cfbf265d19016e1c92e78914564736f6c63430008150033";
const isSuperArgs = (xs) => xs.length > 1;
class MerkleProof__factory extends ethers_1.ContractFactory {
    constructor(...args) {
        if (isSuperArgs(args)) {
            super(...args);
        }
        else {
            super(_abi, _bytecode, args[0]);
        }
    }
    getDeployTransaction(overrides) {
        return super.getDeployTransaction(overrides || {});
    }
    deploy(overrides) {
        return super.deploy(overrides || {});
    }
    connect(runner) {
        return super.connect(runner);
    }
    static createInterface() {
        return new ethers_1.Interface(_abi);
    }
    static connect(address, runner) {
        return new ethers_1.Contract(address, _abi, runner);
    }
}
exports.MerkleProof__factory = MerkleProof__factory;
MerkleProof__factory.bytecode = _bytecode;
MerkleProof__factory.abi = _abi;
